### PR TITLE
Fix issue #1851

### PR DESCRIPTION
**Pull request by jan-zajic**
@see http://stackoverflow.com/questions/20065304/what-is-the-differences-between-begin-rsa-private-key-and-begin-private-key

I added server-key-pkcs1.pem to test sources/tls, which is server-key.pem in the same place converted using command:
openssl rsa -inform PEM -outform PEM -in server-key.pem -out server-key-pkcs1.pem

Test fail before fix:
java.lang.RuntimeException: Missing -----BEGIN PRIVATE KEY----- delimiter
	at io.vertx.core.net.impl.KeyStoreHelper.loadPems(KeyStoreHelper.java:271)
	at io.vertx.core.net.impl.KeyStoreHelper.access$0(KeyStoreHelper.java:246)
	at io.vertx.core.net.impl.KeyStoreHelper$KeyCert.loadPrivateKey(KeyStoreHelper.java:212)
	at io.vertx.core.net.impl.KeyStoreHelper$KeyCert.loadStore(KeyStoreHelper.java:202)
	at io.vertx.test.core.KeyStoreTest.testKeyStore(KeyStoreTest.java:394)
	at io.vertx.test.core.KeyStoreTest.testRsaKeyCertPath(KeyStoreTest.java:342)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

Now OK, other tests in same file as well.	
Signed-off-by: jzajic <jan.zajic@corpus.cz>

### CODE REVIEW

**Comment by vietj** [comment]
can you add a step in `ssl.txt` that describes how to generate the key with openssl CLI ?

**Comment by vietj** [comment]
and update the documentation with the supported format + an example

**Comment by jan-zajic** [comment]
@vietj updated. or you mean something else?

**Comment by lulf** [comment]
@vietj  Any update on this? 

The OpenShift mechanisms for generating service certificates output keys in pkcs#1 format. It would be nice to be able to use those with vert.x without having to convert.

File: src/main/java/io/vertx/core/net/impl/KeyStoreHelper.java [hunk 1]
```diff
@@ -242,33 +312,47 @@ public KeyStore loadStore(VertxInternal vertx) throws Exception {
...
-      if (end == -1) {
-        throw new RuntimeException("Missing " + endDelimiter + " delimiter");
-      }
-      String content = pem.substring(index, end);
-      content = content.replaceAll("\\s", "");
-      if (content.length() == 0) {
-        throw new RuntimeException("Empty pem file");
+      
+      String beginDelimiter = beginMatcher.group(1);
+      index += beginMatcher.end();
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/impl/KeyStoreHelper.java]: seems to be a bug

Comment (jan-zajic) [inline src/main/java/io/vertx/core/net/impl/KeyStoreHelper.java]: ?

**Comment by vietj** [comment]
this shall be rather in Vert.x 3.5

@lulf if you need, we can give you code that converts the key from PKCS1 to PKCS8 in the meantime

**Comment by vietj** [comment]
@jan-zajic I've created a rebased version of your branch here (and amended your commits) https://github.com/vietj/vert.x/tree/feature/1802 can you check ?

**Comment by lulf** [comment]
@vietj The problem isn't the conversion so much as the implication of having to rerun the conversion whenever the certificate file is updated due to certificate expiry. I'm using openssl to run the conversion for now.

**Comment by vietj** [comment]
I mean that the conversion could be done with java, in your app until Vert.x supports this feature.

**Comment by vietj** [comment]
@jan-zajic I've reworked your patch so it could be handled by the IP process due to the fork of the existing source key parser file and now it is in master - thanks for the contribution

@lulf would you mind to give a try ?

**Comment by lulf** [comment]
It is able to read the cert now @vietj , thanks !

**Pull request closed without merging**